website/blog/2018-07-19-whats-happening-with-the-pipeline-proposal.md

```diff
@@ -149,7 +149,7 @@ promise |> await #;
 
 If there are any operators, parentheses (including for method calls), brackets, or anything other than identifiers and dot punctuators, then a topic token is necessary. This avoids footguns and eliminate ambiguity when not using a topic token.
 
-Smart pipelines thus resolve the issue of async in an integrative way, allowing all possible expressions to be emedded in a pipeline; not only `await`, but also `typeof`, `yield`, and another other operator desired.
+Smart pipelines thus resolve the issue of async in an iterative way, allowing all possible expressions to be embedded in a pipeline; not only `await`, but also `typeof`, `yield`, and another other operator desired.
```

Comment:
I'm not sure `iterative` is the right way. I changed a similarly misspelled word to `integrate`...

Reply:
cc @mAAdhaTTah

Reply:
In this case, no, "integrative" was intended. The idea is the Smart Pipelines solves async by solving all expressions in pipelines (vs. F#-style, which has one-off special handling for `await`). "Holistic" could be an alternative word choice, or we could rework this sentence so it's clearer, but "iterative" is not the intended meaning here.

Reply:
Would `integrated` work?

I think I'll go with `holistic`, that at least makes some sense, although it'd be nice if you could explain a bit more here.

Reply:
To my ear, "integrative" sounded better, but I wasn't really sure why. I checked out the definitions of each:


>integrated | ˈin(t)əˌɡrādəd |
>adjective
>with various parts or aspects linked or coordinated: an integrated and high-quality public transportation system.


>integrative | ˈintəɡrādiv |
>adjective
>serving or intending to unify separate things: an integrative approach to learning.

I _think_ "integrative" matches better what I'm trying to get at here. There are several operators that can be embedded into a Smart pipeline. Instead of solving for each of them individually, we solve for them by solving "expressions in the pipeline" as a whole, hence "unify[ing] separate things." They're not so much "linked or coordinated" –they're not working with each other, exactly, so much as being handled as a unified whole, as expressions.

This is in contrast with F#, which solves it with a one-off implementation for `await`.  I always considered the `await` usage in F# to be as if you were treating it like a unary function, even though that's not really what it is, but it doesn't treat expressions as a unified hole.

Holistic kinda works too:

>holistic | hōˈlistik |
>adjective
>chiefly Philosophy characterized by comprehension of the parts of something as intimately interconnected and explicable only by reference to the whole.

Smart treats all expressions as a single whole, rather than F# treating await like a single component part. That said, having read the definitions, I think "integrative" makes the most sense, but given that I had to look it up, it's not really easing the clarity of my communication here. We could perhaps rewrite this graf to be something like:

>Smart pipelines thus resolve the issue of async by solving for expressions as a whole, enabling any of them to be embedded in a pipeline; not only `await`, but also `typeof`, `yield`, and any other operator desired.

Thoughts?

Reply:
I'm fine with leaving it `integrative`

Reply:
Dropped. It's odd, I don't think I've ever encountered this word. I did a search w/ google and it seems to be mostly used in Health/Medicine.